pkg/allowedpaths/cmd.go

```diff
@@ -0,0 +1,784 @@
+package allowedpaths
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"runtime"
+)
+
+type AllowedCommand func(ctx context.Context, arg ...string) (*exec.Cmd, error)
+
+func Airport(ctx context.Context, arg ...string) (*exec.Cmd, error) {
+	if runtime.GOOS != "darwin" {
+		return nil, errors.New("airport supported on darwin only")
+	}
+
+	fullPathToCmdValidated, err := validatedPath("/System/Library/PrivateFrameworks/Apple80211.framework/Versions/Current/Resources/airport")
+	if err != nil {
+		return nil, fmt.Errorf("airport not found: %w", err)
+	}
+	return newCmd(ctx, fullPathToCmdValidated, arg...), nil
```

Comment:
I would probably store an array of possible paths, and make this more of a helper. I think that'll give us a single place to adjust nix (or similar).

Reply:
I think the single place to adjust for nix will already be in `validatedPath`. I liked this approach because it made it easier to adjust for one-offs (e.g. like echo, which doesn't have a path on Windows). But it does certainly feel very repetitive.

Reply:
Oh, I think I misunderstood -- if you mean combining so that we have a function like --

```
func newValidatedCmd(ctx context.Context, knownPath string, arg ...string) *exec.Cmd {
        // do all the validatePath stuff here
	return exec.CommandContext(ctx, fullPathToCmd, arg...) //nolint:forbidigo
}
```

then sure, that seems fine to me

Reply:
I'm not even sure what I'm suggesing. 😆 

This blurs with the func vs interface conversation. I'm thinking about how to represent things. We have a short name, a set of possible paths, maybe some extra logic. I can imaging hanging that on an object with some interface glued on, or what you have now. I guess it's pretty easy to swap back and forth.

But I was chewing on something like:
```
type knownCommand struct {
  possiblePaths []string
}

func (kc knownCommand) Ctx(ctx context.Context, args ...string) (exec.Command, error) {
 ...
}

var Airport = knownCommand{
  possiblePaths: []string{"/System/Library/PrivateFrameworks/Apple80211.framework/Versions/Current/Resources/airport"}
}
```